// @connectv/marked/dist/es5/defaults/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var space_1 = require("./space");
var hr_1 = require("./hr");
var heading_1 = require("./heading");
var code_1 = require("./code");
var table_1 = require("./table");
var block_quote_1 = require("./block-quote");
var paragraph_1 = require("./paragraph");
var list_1 = require("./list");
var html_1 = require("./html");
function fill(options) {
    return {
        Space: options.Space || space_1.Space,
        Hr: options.Hr || hr_1.Hr,
        Heading: options.Heading || heading_1.Heading,
        Code: options.Code || code_1.Code,
        Table: options.Table || table_1.Table,
        TableRow: options.TableRow || table_1.TableRow,
        TableCell: options.TableCell || table_1.TableCell,
        TableHeaderCell: options.TableHeaderCell || table_1.TableHeaderCell,
        BlockQuote: options.BlockQuote || block_quote_1.BlockQuote,
        List: options.List || list_1.List,
        ListItem: options.List || list_1.ListItem,
        Html: options.Html || html_1.Html,
        Paragraph: options.Paragraph || paragraph_1.Paragraph,
    };
}
exports.fill = fill;
